code/game/machinery/computer/computer.dm

```diff
@@ -94,9 +97,9 @@
 	..()
 	update_icon()
 	if(!(stat & (BROKEN|NOPOWER)))
-		SetLuminosity(2)
+		set_light(2)
 	else
-		SetLuminosity(0)
+		set_light(light_range_on, light_power_on)
```

Comment:
You've muddled it up.
This should be set_light(0), and the above set should be (light_range_on, light_power_on), because that's for if the machine is working, and this is for if it's not.